src/main/java/ch/naviqore/service/Connection.java

```diff
@@ -1,30 +1,19 @@
 package ch.naviqore.service;
 
-import org.jetbrains.annotations.NotNull;
-import org.jetbrains.annotations.Nullable;
-
 import java.util.List;
 
+/**
+ * A connection in a public transit schedule, consisting of multiple public transit legs and walks.
+ */
 public interface Connection {
-    @NotNull List<Leg> getLegs();
-
-    @NotNull Location getStartLocation();
-
-    @NotNull Location getEndLocation();
-
-    @Nullable Stop getStartStop();
-
-    @Nullable Stop getEndStop();
-
-    @NotNull ArrivalTime getArrivalTime();
-
-    @NotNull DepartureTime getDepartureTime();
 
-    int getDuration();
+    List<PublicTransitLeg> getPublicTransitLegs();
```

Comment:
I'm not sure this is a good solution, as maybe at a later stage you would have a third (or more) type of travel mode, e.g. E-Scooter, Taxi, etc. How about adding an enum LegType and a method getType()? This would make them easy to filter etc. and I think the most typical usage will anyway be looping leg by leg and building something in chronological order. Which will be more complicated with 2 or more lists.

Reply:
Done in [c5dbe98](https://github.com/naviqore/public-transit-service/pull/21/commits/c5dbe989eb6181e05a63fab3105b7092323eb40f)